authentication/azure/auth.go

```diff
@@ -129,6 +129,23 @@ func (s EnvironmentSettings) GetTokenCredential() (azcore.TokenCredential, error
 		}
 	}
 
+	// 4. MSI using Default Credential
+	// DefaultAzureCredential is a default credential chain for applications that will be deployed to Azure.
+	// It combines credentials suitable for deployed applications with credentials suitable in local development.
+	// It attempts to authenticate with each of these credential types, in the following order:
+	//	- EnvironmentCredential
+	//	- ManagedIdentityCredential
+	//	- AzureCLICredential
+	// Consult the documentation for these credential types for more information on how they authenticate.
+	{
+		cred, err := azidentity.NewDefaultAzureCredential(nil)
```

Comment:
@berndverst can you review this `auth` part please?

Reply:
or @ItalyPaleAle

Reply:
This does not look right at all. I understand what the DefaultAzureCredential does but you just duplicated a bunch of things the code already supports today.

Whether we should the Azure CLI credential here needs to be discussed, but if so that should be the only change -- adding support for the Azure CLI credential.

@ItalyPaleAle knows this code much better so he should review this.